### DISCUSSION TITLE
Managing thread priority [13594]

### DISCUSSION THREAD

**Discussion by bsoe-flexiv** [eProsima/Fast-DDS#2435] [Ideas]
When I run a publisher and subscriber, fast-dds launches multiple threads. This happens even when both listeners are set to null. Currently, I am setting the process priority, before creating my publisher and subscriber, so that all the fast-dds thread inherit this priority.

What is the best way to:
1. minimize the number of threads launched by fast-dds?
2. set the thread priorities of threads launched by fast-dds, for optimal realtime performance?

**Comment by MiguelCompany** [comment]
The threads spawned by Fast DDS are enumerated in the docs [here](https://fast-dds.docs.eprosima.com/en/latest/fastdds/library_overview/library_overview.html?highlight=threads#programming-and-execution-model)

Allowing the user more control over the number of threads and their scheduling parameters is on our roadmap, but not possible at the moment.

**Comment by bsoe-flexiv** [comment]
@MiguelCompany, thanks for the reply and contributing this amazing library. I am happy to hear thread management is on the roadmap :)

**Comment by Linkeway** [comment]
Hi, I've been thinking about this workaround. Any idea?
```c++
std::make_shared<SubListener>(
      [](...) {
        static thread_local bool thread_is_set = pthread_setschedparam(pthread_self()...);
         // actual callback below...
      });
```

**Comment by MiguelCompany** [comment]
@briansoe66 It's been a long time, but we have just merged #4013, enabling the configuration of certain parameters for all the threads spawned by Fast DDS.

You can find relevant documentation [here](https://fast-dds.docs.eprosima.com/en/latest/fastdds/library_overview/library_overview.html#programming-and-execution-model) and [here](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/core/policy/eprosimaExtensions.html#threadsettingsqos)
